### ISSUE TITLE
bootstrap.sh: Syntax error: "(" unexpected

### ISSUE THREAD

**Issue by DaVilla7** [romulogcerqueira/sonar_simulation-buildconf#7]
Hello! thank you for your open source code, there is a unexpected error when I execute the bootsrtap.sh file, the log was illustrated as follows
```root@mqxwd68-System-Product-Name:/home/mqxwd68/Downloads/sonar_simulation# sh bootstrap.sh
autoproj_install:53: warning: Insecure world writable dir /home/mqxwd68/Downloads/sonar_simulation/install/bin in PATH, mode 040777
Detected 'gem' to be /usr/bin/gem2.3
Detected bundler at /root/.local/share/autoproj/gems/ruby/2.3.0/bin/bundle
Installing autoproj in /root/.local/share/autoproj/gems/ruby/2.3.0
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Using rake 12.3.1
Using equatable 0.5.0
Using tty-color 0.4.3
Using pastel 0.7.2
Using tty-cursor 0.5.0
Using necromancer 0.4.0
Using hitimes 1.3.0
Using timers 4.1.2
Using tty-screen 0.6.5
Using wisper 2.0.0
Using tty-reader 0.2.0
Using tty-prompt 0.15.0
Using facets 3.1.0
Using utilrb 3.0.1
Using autobuild 1.15.0
Using backports 3.11.4
Using bundler 1.16.4
Using concurrent-ruby 1.0.5
Using ffi 1.9.25
Using rb-inotify 0.9.10
Using thor 0.20.0
Using tty-spinner 0.8.0
Using xdg 2.2.3
Using autoproj 2.8.7
Bundle complete! 2 Gemfile dependencies, 24 gems now installed.
Bundled gems are installed into `/root/.local/share/autoproj/gems`
starting the newly installed autoproj for stage2 install
saving temporary env.sh and .autoproj/env.sh
running 'autoproj envsh' to generate a proper env.sh
  updated environment
running 'autoproj osdeps' to re-install missing gems
  installing OS packages: ruby-thor
  bundler: connected to https://rubygems.org/
  environment already up-to-date
Command finished successfully at 2018-09-07 17:33:51 +0800 (took 6 secs)
bootstrap.sh: 4: /home/mqxwd68/Downloads/sonar_simulation/base/scripts/shell/bash: Syntax error: "(" unexpected
```

Could you please tell me what's the matter with this source code?
by the way, my environment is **ubuntu16.4+ruby2.3.1**

**Comment by romulogcerqueira** [comment]
I have never seen this error before. Could you repeat the installation and output it here?

**Comment by chyphen777** [comment]
It got the same error.  I redid installation from the scratch, and the error was gone, while I got other error. It seems the error occurs when "sh bootstrap.sh" is run more than once.  So on a different, but somewhat related to" issue, how do I rebuild the package after making some change (i.e. errors in typelib with a different version of boost?).  Thanks. CC.
